tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/upgrade/ACSEnv.java

```diff
@@ -0,0 +1,291 @@
+package org.alfresco.elasticsearch.upgrade;
+
+import static java.time.Duration.ofMinutes;
+
+import static org.alfresco.elasticsearch.upgrade.Utils.waitFor;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.alfresco.elasticsearch.upgrade.AvailabilityProbe.ProbeResult;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.images.builder.Transferable;
+
+class ACSEnv implements AutoCloseable
+{
+    private final Config cfg;
+    private final GenericContainer<?> alfresco;
+    private final GenericContainer<?> postgres;
+    private final List<GenericContainer<?>> createdContainers = new ArrayList<>();
+
+    private RepoHttpClient repoHttpClient;
+
+    private String metadataDumpToRestore;
+    private Path alfDataHostPath;
+    private boolean readOnlyContentStore;
+
+    public ACSEnv(Config cfg, final Network network, final String indexSubsystemName)
+    {
+        this.cfg = cfg;
+        postgres = newContainer(PostgreSQLContainer.class, cfg.getPostgreSQLImage())
```

Comment:
I personally would do something like "startEnvironmentContainers()" and within it "startPostgre", "startActiveMq", "startSharedFileStore" etc. It is understandable now though, so your choice.

Reply:
I moved containers creation to separate methods.